### PR TITLE
Fixed test cases

### PR DESCRIPTION
**Pull request by luiss07**
I updated the tests according to the protocol as reported in #118.

Additionally, I added the channels to handle DroneEvents which were uninitialized and could have resulted in the tests not working (as pointed out on telegram).

Let me know if you think the fixes seem reasonable.

> Note: I had to add the wg_controller as the test dependency since it was not used before. And some other minor adjustments.

### CODE REVIEW

**Comment by matteo-dinoia** [comment]
Also change all recv to recv_timeout. Or the test will take 5 second each to fail. 
The rest of the test, i'll look later. But i think we are trying to solve the problem of a broken protocol with test that acts as protocols.

**Comment by luiss07** [comment]
> Also change all recv to recv_timeout. Or the test will take 5 second each to fail. The rest of the test, i'll look later. But i think we are trying to solve the problem of a broken protocol with test that acts as protocols.

Do you mean that you want me to put less than 5 seconds? I don't get the second sentence

File: crates/wg_tests/src/test_fragments.rs [hunk 1]
```diff
@@ -86,25 +92,29 @@ pub fn generic_fragment_drop<T: Drone + Send + 'static>() {
...
-    };
-    let nack_packet = Packet {
-        pack_type: PacketType::Nack(dropped),
-        routing_header: srh,
-        session_id: 1,
-    };
+    let nack_packet = Packet::new_nack(
+        SourceRoutingHeader {
+            hop_index: 1,
+            hops: vec![11, 1],
```
Comment (matteo-dinoia) [inline crates/wg_tests/src/test_fragments.rs:98]: Should be 12, 11 1

**Comment by matteo-dinoia** [comment]
> > Also change all recv to recv_timeout. Or the test will take 5 second each to fail. The rest of the test, i'll look later. But i think we are trying to solve the problem of a broken protocol with test that acts as protocols.
> 
> Do you mean that you want me to put less than 5 seconds? I don't get the second sentence

Yeah like 300 ms is way more than enough. 
The second is about the fact we are trying to fix edge case by creating a single test suite instead of a working protocol.

File: crates/wg_tests/src/test_fragments.rs (hunk 1 above)
Comment (luiss07) [reply crates/wg_tests/src/test_fragments.rs:98]: Are you sure? In this case the packet is being dropped by 11, it did not reach 12

File: crates/wg_tests/src/test_fragments.rs [hunk 2]
```diff
@@ -86,25 +92,29 @@ pub fn generic_fragment_drop<T: Drone + Send + 'static>() {
...
+            nack_type: NackType::Dropped,
+        },
+    );
 
     // Client listens for packet from the drone (Dropped Nack)
     assert_eq!(c_recv.recv().unwrap(), nack_packet);
+    // SC listen for event from the drone
+    assert_eq!(
+        d_event_recv.recv().unwrap(),
+        DroneEvent::PacketDropped(nack_packet)
```
Comment (matteo-dinoia) [inline crates/wg_tests/src/test_fragments.rs:112]: It should be the packet dropped not the nack. We should also decide when sending the nack we should also send a drone event of packetSend

**Comment by matteo-dinoia** [comment]
In general, just the approach we are taking but at the moment let's just merge this PR.

**Comment by luiss07** [comment]
> In general, just the approach we are taking but at the moment let's just merge this PR.

Added the `recv_timeout`. The last thing to verify is whether the the Nack that you pointed out in the unresolved conversation should contain 12 or not.

**Comment by matteo-dinoia** [comment]
Yeah sorry i got it wrongly. Merging.
